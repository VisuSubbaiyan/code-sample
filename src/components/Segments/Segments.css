.segments {
  background: #eeeeee;
  padding: 20px;
  height: 100vh;
}

.sub-heading {
  margin: 0;
  padding: 10px 0;
}

.segment {
  text-align: left;
  border-left: 2px solid #06A2F5;
}

.segment .time {
  padding: 15px 10px;
  margin: 0;
}

.no-top-padding .time {
  padding-top: 0;
}

/* .segment.no-padding .time {
  padding-top: 0;
} */

.segment .time::before {
  height: 12px;
  width: 12px;
  margin-left: -17px;
  background: #06A2F5;
  border-radius: 50%;
  position: absolute;
  content: '';
}

.segment-active {
  border-color: #05164d;
}
.segment-active .time::before {
  height: 12px;
  width: 12px;
  margin-left: -17px;
  background: #05164d;
  border-radius: 50%;
  position: absolute;
  content: '';
}

.card {
  padding: 16px;
  background: #06A2F5;
  color: #05164d;
  display: flex;
}

.segment-active .card {
  background: #05164d;
  color: #eeeeee;
}

.card .title {
  padding-left: 10px;
}

.card .terminal-details {
  padding-top: 10px;
}

.card .status-busy {
  padding-left: 10px;
  color: #ff0000;
}

.card .right-segment {
  width: 60%;
}

.card .fa {
  font-size: 24px;
}

.highlight {
  color: #ffad00;
}