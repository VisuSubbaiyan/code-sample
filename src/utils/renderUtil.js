import React from 'react';

import {Airport, BoardingPass, Ground} from '../components';

export const renderItems = items => items.map(item => {
  const { type, segment_id } = item;

  switch (type) {
    case 'GROUND': {
      return <Ground {...item} key={segment_id} className="segment-active no-top-padding" />;
    }
    
    case 'AIRPORT': {
      return <Airport {...item} key={segment_id} />;
    }

    default:
      return <BoardingPass {...item} key={segment_id} />
  }
});
