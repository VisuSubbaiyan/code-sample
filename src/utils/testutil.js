import { mount, shallow } from 'enzyme';
import { prettyPrint } from 'html';

export const mockFunction = jest.fn(value => value);

export const sampleFriend = {
  name: 'Sample friend',
  gender: 'male'
};

export const sampleId = 1;

export const createMountedComponent = ({ children, isShallow }) => {
  const mountedComponent = isShallow ? shallow(children) : mount(children);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

