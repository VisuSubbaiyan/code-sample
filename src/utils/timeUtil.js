import moment from 'moment';

export const formatTime = time => {
  const tempTime = moment.duration(time);
  
  return `${tempTime.hours()}: ${tempTime.minutes()}`;
}