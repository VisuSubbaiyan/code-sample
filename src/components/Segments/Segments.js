import React, { Component } from 'react';
// import PropTypes from 'prop-types';

import { renderItems } from '../../utils/renderUtil.js';

import data from '../../data.json';
import './Segments.css';

class Segments extends Component {
  state = {...data}

  render() {
    const { segments, origin_iata, destination_iata} = this.state;
    return (
      <div className="segments">
        <h2 className="sub-heading">{`${origin_iata} - ${destination_iata}`}</h2>
        {renderItems(segments)}
      </div>
    )
  }
}

export default Segments;
