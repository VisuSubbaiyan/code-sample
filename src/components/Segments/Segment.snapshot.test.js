import React from 'react';

import { createMountedComponent } from '../../utils/testutil';

import Segments from './Segments';


describe('<Ground /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <Segments />
  });

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
  })
});