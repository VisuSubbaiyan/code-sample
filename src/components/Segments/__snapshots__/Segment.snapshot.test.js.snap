// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ground /> should render correctly according to props 1`] = `
"<div class=\\"segments\\">
  <h2 class=\\"sub-heading\\">TXL - MUC</h2>
  <div class=\\"segment segment-active no-top-padding\\">
    <h3 class=\\"time\\">15: 13</h3>
    <div class=\\"card\\">
      <div><i class=\\"fa fa-bus\\"></i>
      </div>
      <div class=\\"title\\">
        <div>Pick up from</div><span class=\\"highlight\\">Alte Schönhauser Straße 3, 10119 Berlin</span>
      </div>
    </div>
  </div>
  <div class=\\"segment\\">
    <h3 class=\\"time\\">15: 13</h3>
    <div class=\\"card\\">
      <div><i class=\\"fa fa-plane-departure\\"></i>
      </div>
      <div class=\\"title\\">
        <div><span class=\\"highlight\\">TXL</span> Airport</div>
        <div class=\\"terminal-details\\"><span class=\\"highlight\\">Terminal C </span><span class=\\"status-busy\\">Busy</span>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"segment\\">
    <h3 class=\\"time\\">15: 13</h3>
    <div class=\\"card boarding-pass\\">
      <div class=\\"flight-details\\">
        <div>LH2031</div>
        <div class=\\"boarding-time\\">
          <h3 class=\\"time\\">15: 13</h3>
        </div>
      </div>
      <div class=\\"boarding-details\\">
        <h2>TXL</h2>
        <div><i class=\\"fa fa-plane\\"></i>
        </div>
        <h2>MUC</h2>
      </div>
      <div class=\\"gate-details\\"><span>Gate C12</span><span>Seat 12C</span><span class=\\"ontime\\">OnTime</span>
      </div>
    </div>
  </div>
</div>"
`;
